server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -4931,6 +4936,15 @@ public UserVm createVirtualMachine(DeployVMCmd cmd) throws InsufficientCapacityE
             addExtraConfig(vm, caller, extraConfig);
         }
 
+        if (cmd.getCopyImageTagsToVm()) {
+            final ResourceTag.ResourceObjectType templateType = (_templateDao.findById(templateId).getFormat() == ImageFormat.ISO) ? ResourceTag.ResourceObjectType.ISO : ResourceTag.ResourceObjectType.Template;
```

Comment:
Potential NPE case, please make the code defensive:
- Find the template/iso using the templateId
- Check for nullity and then proceed with the logic/code